src/Microsoft.OData.Core/UriParser/Resolver/UnqualifiedODataUriResolver.cs

```diff
@@ -47,27 +47,39 @@ public override IEnumerable<IEdmOperation> ResolveBoundOperations(IEdmModel mode
                 return base.ResolveBoundOperations(model, identifier, bindingType);
             }
 
-            return FindAcrossModels<IEdmOperation>(model, identifier, this.EnableCaseInsensitive)
-                .Where(operation =>
-                    operation.IsBound
-                    && operation.Parameters.Any()
-                    && operation.HasEquivalentBindingType(bindingType));
+            return FindOperations(model, identifier, this.EnableCaseInsensitive, true, bindingType);
         }
 
-        private static IEnumerable<T> FindAcrossModels<T>(IEdmModel model, String qualifiedName, bool caseInsensitive) where T : IEdmSchemaElement
+        private static IEnumerable<IEdmOperation> FindOperations(IEdmModel model, string qualifiedName, bool caseInsensitive, bool isBound = false, IEdmType bindingType = null)
         {
-            Func<IEdmModel, IEnumerable<T>> finder = (refModel) =>
-                refModel.SchemaElements.OfType<T>()
-                .Where(e => string.Equals(qualifiedName, e.Name, caseInsensitive ? StringComparison.OrdinalIgnoreCase : StringComparison.Ordinal));
+            IList<IEdmOperation> results = new List<IEdmOperation>();
+
+            StringComparison strComparison = caseInsensitive ? StringComparison.OrdinalIgnoreCase : StringComparison.Ordinal;
 
-            IEnumerable<T> results = finder(model);
+            GetOperationsForModel(model, qualifiedName, results, isBound, bindingType, strComparison);
 
             foreach (IEdmModel reference in model.ReferencedModels)
             {
-                results.Concat(finder(reference));
+                GetOperationsForModel(reference, qualifiedName, results, isBound, bindingType, strComparison);
             }
 
             return results;
         }
+
+        private static void GetOperationsForModel(IEdmModel model, string qualifiedName, IList<IEdmOperation> results, bool isBound, IEdmType bindingType, StringComparison strComparison)
+        {
+            foreach (IEdmSchemaElement schemaElement in model.SchemaElements)
+            {
+                if (schemaElement is IEdmOperation operation)
+                {
+                    if (((!isBound && !operation.IsBound) || (isBound && operation.IsBound && operation.Parameters.Any() && operation.HasEquivalentBindingType(bindingType))) &&
+                        string.Equals(qualifiedName, operation.Name, strComparison))
+                    {
+                        results.Add(operation);

```

Comment:
if the result is an IEnumerable<IEdmOperation>, why are we building a list at all, rather than just yielding here?

the list obviously works better for caching, but it's unclear at this point where the right place would be to cache.

Reply:
I would say there was merit in the original PR, but after the recent changes I agree that this should not be `List` anymore and just use `yield return`